### ISSUE TITLE
[BUG] Hermitian gradient

### ISSUE THREAD

**Issue by cvqmm** [PennyLaneAI/pennylane#4864]
### Expected behavior

the gradient of an expectation value of a Hermitian with respect to the Hermitian entries is wrong

### Actual behavior

it returns a tensor of zeros

### Source code

```python
import pennylane as qml
from pennylane import numpy as np, jacobian
dev=qml.device("default.qubit", wires=1)
@qml.qnode(dev)
def circuit(h):
    return qml.expval(qml.Hermitian(h, wires=range(1)))

h=np.random.rand(2,2)
h+=h.T
jacobian(circuit)(h)
```

### Tracebacks

```python
UserWarning: Output seems independent of input.
  warnings.warn("Output seems independent of input.")
array([[0., 0.],
       [0., 0.]])
```

### System information

```shell
Name: PennyLane
Version: 0.29.1
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /fs/home/cvqmm/.conda/envs/qcware/lib/python3.8/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, retworkx, scipy, semantic-version, toml
Required-by: covvqetools, PennyLane-Lightning, PennyLane-Qchem

Platform info:           Linux-4.18.0-372.32.1.el8_6.x86_64-x86_64-with-glibc2.10
Python version:          3.8.13
Numpy version:           1.23.0
Scipy version:           1.8.1
Installed devices:
- lightning.qubit (PennyLane-Lightning-0.30.0)
- default.gaussian (PennyLane-0.29.1)
- default.mixed (PennyLane-0.29.1)
- default.qubit (PennyLane-0.29.1)
- default.qubit.autograd (PennyLane-0.29.1)
- default.qubit.jax (PennyLane-0.29.1)
- default.qubit.tf (PennyLane-0.29.1)
- default.qubit.torch (PennyLane-0.29.1)
- default.qutrit (PennyLane-0.29.1)
- null.qubit (PennyLane-0.29.1)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by albi3ro** [comment]
Thanks for posting the issue @cvqmm . I've opened a Pull Request that will hopefully be fixing this problem. This should make it to master soon and will come out with the next release in early January.

**Comment by albi3ro** [comment]
Closed by #4866
